Mechanics Tutorials (#18828)
idaholab/moose: modules/tensor_mechanics/doc/content/modules/tensor_mechanics/tutorials/introduction/step02.md

# Step 2 - Adding boundary conditions

In the previous step we set up a basic small strain mechanics simulation that
did... nothing. In this step we're adding some loading on the top and we'll fix
the displacements on the bottom surface of our block.

!listing modules/tensor_mechanics/tutorials/introduction/step02.i

## Input file

### `BCs`

BCs stands for *boundary conditions*. Those apply to the boundaries (or
sidesets) of the simulation domain. In all boundary condition objects you will
see the mandatory `boundary` parameter, which expects a list of sideset names or
IDs.

#### `DirichletBC`

> The two [`DirichletBC`](DirichletBC.md) boundary conditions are bot set on the

Reviewer: ```suggestion
The two [`DirichletBC`](DirichletBC.md) boundary conditions are both set on the
```

> *bottom* surface of the simulation domain. The fix the `disp_x` and `disp_y`

Reviewer: ```suggestion
*bottom* surface of the simulation domain. This fixes the `disp_x` and `disp_y`
```

Reviewer: or sets

```

### `Preconditioning`

The [`[Preconditioning]`](syntax/Preconditioning/index.md) block lets the user specify the which parts of the Jacobian matrix are built. Here we're selecting the [single matrix preconditioner](SingleMatrixPreconditioner.md) with the [!param](/Preconditioning/SMP/full) option set to `true` to build a fully coupled Jacobian matrix. This helps the solver to better take the cross coupling between displacement variables into account and will lead to improved convergence.

### `Executioner`

Using the [!param](/Executioner/Transient/petsc_options_iname) and
[!param](/Executioner/Transient/petsc_options_value) parameters we can specify
pairs of PETSc options and their values. `-pctype lu` selects a direct LU
decomposition solver. It is a good choice for small problems  but does not provide
a lot of scalability for larger problems.

## Questions

Go ahead and run the input and visualize the result. Look at how the applied
boundary condition effect the deformation of the sample.

### Exploring parameters

> Experiment with different settings for the mechanical properties of the sample
> and the applied loading. What happens if you drastically reduce the Young's

> > modulus or increase the applied pressure. Is the simulaion result still valid?

Reviewer: ```suggestion
> modulus or increase the applied pressure. Is the simulation result still valid?
```
